spring-batch-core/src/main/java/org/springframework/batch/core/repository/dao/JdbcJobInstanceDao.java

```diff
@@ -301,11 +302,26 @@ public int getJobInstanceCount(@Nullable String jobName) throws NoSuchJobExcepti
 	 *
 	 * @param jobIncrementer
 	 *            the {@link DataFieldMaxValueIncrementer}
+	 *
+	 * @deprecated as of v5.0 in favor of using the {@link setJobInstanceIncrementer}
 	 */
+	@Deprecated
 	public void setJobIncrementer(DataFieldMaxValueIncrementer jobIncrementer) {
 		this.jobIncrementer = jobIncrementer;
 	}
 
+	/**
+	 * Setter for {@link DataFieldMaxValueIncrementer} to be used when
+	 * generating primary keys for {@link JobInstance} instances.
+	 *
+	 * @param jobIncrementer the {@link DataFieldMaxValueIncrementer}
+	 *
+	 * @since 5.0
+	 */
+	public void setJobInstanceIncrementer(DataFieldMaxValueIncrementer jobIncrementer) {
+		this.jobIncrementer = jobIncrementer;
```

Comment:
The parameter and field could also be renamed to `jobInstanceIncrementer` here. I will do that on merge.